tests/test_utils/test_check_funcions.py

```diff
@@ -80,3 +83,17 @@ def test_resource_checkPK(self, test_input: dict):
             error = error[0:10]
 
         assert error == test_input["expected"]
+
+
+    @pytest.mark.parametrize("test_input, test_expected, test_result", [(dataset.headers, {"spatial_code", "spatial_type", "spatial_date",}, False)])
+    def test_check_missing_fields(self, test_input, test_expected, test_result):
+        """check all expected items exist in fields: return False"""
+        assert  check_missing_fields(test_input, test_expected) == test_result
+
+    @pytest.mark.parametrize("test_input, test_expected, test_result", [(dataset.headers, {"test_field", "spatial_type", "spatial_date",}, "Missing column(s) ['test_field'].")])
+    def test_check_missing_fields_found(self, test_input, test_expected, test_result):
+        """ check not all expected items exist in fields: return error with printed list with missing items 
+        return error string message """
+
+        result = check_missing_fields(test_input, test_expected)
+        assert result[0:33] == test_result
```

Comment:
Je `assert` hier de helft van de foutmelding. Ik zou het netter vinden om het hele ding uit te schrijven, en te vergelijken met `test_result`. Dat is ook leesbaarder dan een string index. Iets als:

```
assert "Missing column(s) ['test_field']. Mandatory fields are: ['test_field', 'spatial_type', 'spatial_date']" == test_result
```

Reply:
de rest van de error code is niet relevant voor de test: er wordt een error gegeven en die geeft de 'missings fields' terug.